extensions/objects/templates/position-of-terms-editor.component.ts

```diff
@@ -19,32 +19,43 @@
 // Every editor directive should implement an alwaysEditable option. There
 // may be additional customization options for the editor that should be passed
 // in via initArgs.
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { AppConstants } from 'app.constants';
 
-angular.module('oppia').component('positionOfTermsEditor', {
-  bindings: {
-    value: '='
-  },
-  template: require('./position-of-terms-editor.component.html'),
-  controller: ['POSITION_OF_TERMS_MAPPING',
-    function(POSITION_OF_TERMS_MAPPING) {
-      const ctrl = this;
+export interface PositionOfTerm {
+  readonly name: 'string';
+  readonly humanReadableName: 'string';
+}
 
-      ctrl.onChangePosition = function() {
-        ctrl.value = ctrl.localValue.name;
-      };
+@Component({
+  selector: 'position-of-terms-editor',
+  templateUrl: './position-of-terms-editor.component.html',
+  styleUrls: []
+})
+export class PositionOfTermsEditorComponent implements OnInit {
+  @Input() modalId: symbol;
+  @Input() value;
+  @Output() valueChanged = new EventEmitter();
+  alwaysEditable = true;
+  positionOfTerms = AppConstants.POSITION_OF_TERMS_MAPPING;
+  localValue: PositionOfTerm;
+  constructor() { }
 
-      ctrl.$onInit = function() {
-        ctrl.alwaysEditable = true;
-
-        ctrl.positionOfTerms = POSITION_OF_TERMS_MAPPING;
-
-        ctrl.localValue = ctrl.positionOfTerms[2];
-        for (var i = 0; i < ctrl.positionOfTerms.length; i++) {
-          if (ctrl.positionOfTerms[i].name === ctrl.value) {
-            ctrl.localValue = ctrl.positionOfTerms[i];
-          }
-        }
-      };
+  ngOnInit(): void {
+    this.localValue = this.positionOfTerms[2] as unknown as PositionOfTerm;
```

Comment:
The dropdown should be initialized with "on both sides" but it's empty:
![Screenshot from 2021-04-27 16-15-07](https://user-images.githubusercontent.com/35144226/116229230-df0a3d80-a773-11eb-9c23-1e67e09d8c8f.png)

Reply:
Done.